server/build.gradle

```
@@ -240,3 +254,62 @@ checkstyleContainerTest.onlyIf { "true" != System.getenv("skipContainerTest") }
 spotbugsContainerTest.onlyIf { "true" != System.getenv("skipContainerTest") }
 
 check.dependsOn containerTest, integrationTest, jacocoTestReport
+
+// Task to generate OpenAPI documentation
+tasks.register('generateOpenApiDocs', JavaExec) {
+    group = 'documentation'
+    description = 'Generates OpenAPI documentation files (JSON, YAML, HTML)'
+    
+    dependsOn compileJava
+    classpath = sourceSets.main.runtimeClasspath
+    mainClass = 'org.apache.cassandra.sidecar.docs.OpenApiDocumentationGenerator'
+    
+    def outputDir = layout.buildDirectory.dir('docs/openapi').get().asFile
+    args outputDir.absolutePath
+    
+    outputs.dir outputDir
+    
+    doFirst {
+        if (!outputDir.exists()) {
+            outputDir.mkdirs()
+        }
```

the check is redundant. mkdirs will perform the check itself
```suggestion
        outputDir.mkdirs()
```